postgis/mvt.c

```diff
@@ -721,6 +721,74 @@ LWGEOM *mvt_geom(LWGEOM *lwgeom, GBOX *gbox, uint32_t extent, uint32_t buffer,
 }
 
 /**
+ * Transform a point geometry into vector tile coordinate space.
+ */
+LWGEOM *mvt_geom_point(LWPOINT *lwgeom, GBOX *gbox, uint32_t extent, uint32_t buffer,
+	bool clip_geom)
+{
+	POSTGIS_DEBUG(2, "mvt_geom called");
+	LWGEOM *lwgeom_out = NULL;
+	double width = gbox->xmax - gbox->xmin;
+	double height = gbox->ymax - gbox->ymin;
+	double resx = width / extent;
+	double resy = height / extent;
+	double fx = extent / width;
+	double fy = -(extent / height);
+	double buffer_map_xunits = resx * buffer;
+	double buffer_map_yunits = resy * buffer;
+	const GBOX *ggbox = lwgeom_get_bbox(lwgeom);
+
+	if (width == 0 || height == 0)
+		lwerror("mvt_geom: bounds width or height cannot be 0");
+
+	if (extent == 0)
+		lwerror("mvt_geom: extent cannot be 0");
+
+	if (clip_geom) {
+		GBOX *bgbox = gbox_copy(gbox);
```

Comment:
yes, I missed that one.

A bit off-topic: I was thinking about using a different function that would not mind about clipping and buffers. Going a step further we could use integer arithmetic instead... then the public API would need to change to expose it, and we could not use the generic `ST_AsMVTGeom`.